Microsoft.Azure.Cosmos/src/Serializer/CosmosLinqSerializerOptionsInternal.cs

```diff
@@ -0,0 +1,73 @@
+﻿// ------------------------------------------------------------
+// Copyright (c) Microsoft Corporation.  All rights reserved.
+// ------------------------------------------------------------
+
+namespace Microsoft.Azure.Cosmos.Serializer
+{
+    using System;
+    using Microsoft.Azure.Cosmos;
+
+    /// <summary>
+    /// This class stores user-provided LINQ Serialization Properties.
+    /// </summary>
+    internal sealed class CosmosLinqSerializerOptionsInternal
+    {
+        /// <summary>
+        /// Creates an instance of CosmosSerializationOptionsInternal.
+        /// </summary>
+        public static CosmosLinqSerializerOptionsInternal Create(
+            CosmosLinqSerializerOptions cosmosLinqSerializerOptions, 
+            CosmosSerializer customCosmosSerializer)
+        {
+            switch (cosmosLinqSerializerOptions.LinqSerializerType)
+            {
+                case CosmosLinqSerializerType.Custom:
+                {
+                    if (customCosmosSerializer == null)
+                    {
+                        throw new InvalidOperationException($"Must provide CosmosLinqSerializer if selecting linqSerializerOptions.CustomCosmosLinqSerializer.");
+                    }
+
+                    if (cosmosLinqSerializerOptions.PropertyNamingPolicy != CosmosPropertyNamingPolicy.Default)
+                    {
+                        throw new InvalidOperationException($"CosmosPropertyNamingPolicy must be CosmosPropertyNamingPolicy.Default if selecting linqSerializerOptions.CustomCosmosLinqSerializer.");
+                    }
+
+                    if (customCosmosSerializer is CosmosLinqSerializer customQueryCosmosSerializer)
+                    {
+                        return new CosmosLinqSerializerOptionsInternal(cosmosLinqSerializerOptions, customQueryCosmosSerializer);
+                    }
+
+                    throw new InvalidOperationException($"CosmosSerializer must implement CosmosLinqSerializer if selecting linqSerializerOptions.CustomCosmosLinqSerializer.");

```

Comment:
Isn't the enum called `CosmosLinqSerializerType.Custom`?

```suggestion
                    throw new InvalidOperationException($"CosmosSerializer must implement CosmosLinqSerializer if selecting CosmosLinqSerializerType.Custom.");
```

(same applies to all the other exceptions

Reply:
updated, thanks for catching this